agent/profile

```diff
@@ -1,35 +1,44 @@
 # -*- mode: shell-script -*-
-prefix=/opt/pbench-agent
 
-if [[ ! -f ${prefix}/profile ]]
-then
-    if [[ -f ${PWD}/profile ]]
+function _pbench_pathins() {
+    if [ -d "$1" ] && [[ ":$PATH:" != *"$1:"* ]] ; then
+        PATH="$1${PATH:+:$PATH}"
+    fi
+}
+
+function _pbench_setup() {
+    local prefix=/opt/pbench-agent
+
+    if [[ ! -f ${prefix}/profile ]]
     then
-	prefix=$PWD
-    else
-	# last ditch attempt
-	prefix=$TOP
+	if [[ -f ${PWD}/profile ]]
+	then
+	    prefix=$PWD
+	else
+	    # last ditch attempt
+	    prefix=$TOP
+	fi
     fi
-fi
 
-pathins() {
-    if [ -d "$1" ] && [[ ":$PATH:" != *":$1:"* ]]; then
-        PATH="$1:$PATH"
+    if [[ ! -f ${prefix}/util-scripts/pbench-register-tool-set ]]
+    then
+	echo "*** WARNING *** Something wrong in the pbench base profile:" >&2
+	echo "                Setup has failed - could not determine the proper environment prefix: \"$prefix\"" >&2
+	echo "                You will need to fix it before working with pbench." >&2
+    else
+	local pdir=${prefix}/lib
+	if [[ -z "$PYTHONPATH" || ":$PYTHONPATH:" != *"$1:"* ]]
```

Comment:
Still missing a leading `:`, and it does not seem the `-z` check is necessary.  If `PYTHONPATH` is not set or empty, then the second check will also catch that allowing the `export` to happen.